services/worker/src/worker/dtos.py

```diff
@@ -344,7 +344,7 @@ class DatasetHubCacheResponse(TypedDict):
     preview: bool
     viewer: bool
     partial: bool
-    num_rows: int
+    num_rows: Union[int, None]
```

Comment:
Optional in python means Union[int, None]

```suggestion
    num_rows: Optional[int]
```

Reply:
sure